Backend/src/main/java/ca/mcgill/cooperator/service/EmployerContactService.java

```diff
@@ -151,53 +145,48 @@ public EmployerContact updateEmployerContact(
         if (ec == null) {
             error.append("Employer Contact to update cannot be null! ");
         }
-        if (firstName == null || firstName.trim().length() == 0) {
+        if (firstName != null && firstName.trim().length() == 0) {
             error.append("Employer Contact first name cannot be empty! ");
         }
-        if (lastName == null || lastName.trim().length() == 0) {
+        if (lastName != null && lastName.trim().length() == 0) {
             error.append("Employer Contact last name cannot be empty! ");
         }
-        if (email == null || email.trim().length() == 0) {
+        if (email != null && email.trim().length() == 0) {
             error.append("Employer Contact email cannot be empty! ");
-        } else if (!ServiceUtils.isValidEmail(email)) {
+        } else if (email != null && !ServiceUtils.isValidEmail(email)) {
             error.append("Employer Contact email must be a valid email! ");
         }
-        if (phoneNumber == null || phoneNumber.trim().length() == 0) {
+        if (phoneNumber != null && phoneNumber.trim().length() == 0) {
             error.append("Employer Contact phone number cannot be empty! ");
-        } else if (!ServiceUtils.isValidPhoneNumber(phoneNumber)) {
+        } else if (phoneNumber != null && !ServiceUtils.isValidPhoneNumber(phoneNumber)) {
             error.append("Employer Contact phone number must be a valid number! ");
         }
-        if (company == null) {
-            error.append("Employer Contact company cannot be null! ");
-        }
-        if (employerReports == null) {
-            error.append("Employer Contact employer reports cannot be null! ");
-        }
-        if (coopDetails == null) {
-            error.append("Employer Contact coop details cannot be null!");
-        }
         if (error.length() > 0) {
             throw new IllegalArgumentException(error.toString().trim());
         }
 
-        ec.setFirstName(firstName.trim());
-        ec.setLastName(lastName.trim());
-        ec.setEmail(email.trim());
-        ec.setPhoneNumber(phoneNumber.trim());
-        ec.setEmployerReports(employerReports);
-        ec.setCoopDetails(coopDetails);
-        ec.setCompany(company);
-
-        employerContactRepository.save(ec);
-
-        for (EmployerReport er : employerReports) {
-            er.setEmployerContact(ec);
-            employerReportRepository.save(er);
+        if (firstName != null && firstName.trim().length() > 0) {
```

Comment:
I think you can delete the check for `length() > 0` since we know that it's not 0 if we've hit this if-condition

Reply:
yes